### PR TITLE
Create GitHub action to push to Lokalise

### PR DESCRIPTION
**Pull request by chrisdchow**
The existing Github actions for Lokalise are either outdated or poorly support our use cases. This push action is loosely inspired by https://github.com/marketplace/actions/lokalise-key-push. This existing action is more complex than we need as we only want to push new source language strings/messages. And lacks some necessary functionality like archiving old messages. 

With respect to this image, lifted from FormatJS, this `push action` represents the `Upload Messages` step.
Screen Shot 2021-05-12 at 11 55 18 AM

There will be a separate PR to this repo that adds the `Download Translations` steps.

Additional context:
After contacting Lokalise support, reading their API documentation, and investigating their [CLI](https://github.com/lokalise/lokalise-cli-2-go), I decided that writing this Action with one of their official API clients was the best route. The CLI is very fully featured, however does not support our use case of archiving.

Other references:
1) Action using Lokalise's CLI https://github.com/onin-engineering/lokalise-download-action
2) CrowdIn's office GitHub action using their CLI https://github.com/crowdin/github-action
3) https://github.com/igorDolzh/lokalise-file-push

### CODE REVIEW

File: CODEOWNERS [hunk 1]
```diff
@@ -1 +1,2 @@
 * @actions/actions-runtime
```
Comment (abatilo) [inline CODEOWNERS:1]:
```suggestion
```

We don't want/need another org to be the CODEOWNER

File: CODEOWNERS [hunk 2]
```diff
@@ -1 +1,2 @@
...
+/src @color/tide
```
Comment (abatilo) [inline CODEOWNERS:2]:
```suggestion
* @color/tide @chrisdchow
```

Let's just default this to have you and TIDE own all of it for now.

File: src/lokalise/types.ts [hunk 3]
```diff
@@ -0,0 +1,115 @@
+import { PaginatedResult } from '@lokalise/node-api';
+
+/**
+ * Based on the locale codes used in the file paths.
+ */
+export enum COLOR_LANGUAGE_ISO_CODE {
```
Comment (abatilo) [inline src/lokalise/types.ts:6]: Can we remove the `COLOR` prefix here? This is an open source project so we should not couple our code directly if we can help it.

**Review by abatilo — CHANGES_REQUESTED** [review]
Seems mostly fine overall as a first pass. We should clean up documentation and usage as a fast follow.

Some small requests though.

**Comment by chrisdchow** [comment]
@abatilo thanks for the review! I meant to open this as a draft. I'll address your changes, tidy up the documentation, and finish my PR description 🙈 .

File: README.md [hunk 4]
```diff
@@ -1,105 +1,46 @@
...
-```bash
-$ npm test
-
- PASS  ./index.test.js
-  ✓ throws invalid number (3ms)
-  ✓ wait 500 ms (504ms)
-  ✓ test runs (95ms)
-
-...
 ```
```
Comment (abatilo) [inline README.md:3]:
```suggestion
```yaml
```

**Review by abatilo — CHANGES_REQUESTED** [review]
One quick change for the README. Otherwise LGTM.

**Pull request merged**